service/app.py

```diff
@@ -23,24 +52,76 @@ def send_message():
     # we cannot send render templates as we are not using flask python templates on the frontend
     return jsonify(message="Hello From Backend"), 200
 
-@app.route('/search', methods=['POST'])
-def search():
+@app.route('/register', methods=['POST'])
+def register():
+    if request.method == 'POST':
+        # get JSON data from new user
+        data = request.json
+
+        # get username, password, email, first name, and last name
+        username = data.get('username')
+        password = data.get('password')
+        email = data.get('email')
+        first_name = data.get('firstName')
+        last_name = data.get('lastName')
+
+        logger.info('\nusername: %s', username)
+        logger.info('\npassword: %s', password)
+        logger.info('\nemail: %s', email)
+        logger.info('\nfirst name: %s', first_name)
+        logger.info('\nlast name: %s', last_name)
+
+        # if all credentials are not empty strings, create a new user object, otherwise, throw error
+        if username != '' and password != '' and email != '' and first_name != '' and last_name != '':
+            new_user = User(username=username, password=password, email=email, first_name=first_name, last_name=last_name)
```

Comment:
I think that something like:
```
if all([username, password, email, first_name, last_name]):
```
would be more concise and checks for the same thing, but you could argue that is less explicit

Reply:
I think it is more concise, added it.